BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -323,9 +326,58 @@ public void UpdatePost(Post newPost, Post oldPost)
             SavePost(newPost);
         }
 
-        public bool CheckSlugExists(string slug)
+        public void SaveFiles(List<IFormFile> files)
+        {           
+            foreach(var file in files)
+            {
+                if(file.Length > 0)
+                {                    
+                    using (Stream uploadedFileStream = file.OpenReadStream())
+                    {
+                        byte[] buffer = new byte[uploadedFileStream.Length];

```

Comment:
Should use a reasonable fixed-length buffer.  If I upload a 2GB file, you'd try to allocate a 2GB chunk of memory right here.